crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,451 @@
+    pub const DEFAULT: SizeConstraints = SizeConstraints {
+        min: Size::DEFAULT,
+        suggested: Size::DEFAULT,
+        max: Size::DEFAULT,
+        aspect_ratio: None,
+    };
+
+    /// Creates a new [`SizeConstraints`] with both `max` and `suggested` constraints set to the largest possible extent (100%).
+    pub const FULL: SizeConstraints = SizeConstraints {
+        min: Size::DEFAULT,
+        suggested: Size::FULL,
+        max: Size::FULL,
+        aspect_ratio: None,
+    };
+
+    /// Creates a new [`SizeConstraints`] with the `min`-constraint set to the given values.
+    pub const fn min(width: Val, height: Val) -> SizeConstraints {
+        SizeConstraints {
+            min: Size::new(width, height),
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Creates a new [`SizeConstraints`] with the `suggested`-constraint set to the given values.
+    pub const fn suggested(width: Val, height: Val) -> SizeConstraints {
+        SizeConstraints {
+            suggested: Size::new(width, height),
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Creates a new [`SizeConstraints`] with the `max`-constraint set to the given values.
+    pub const fn max(width: Val, height: Val) -> SizeConstraints {
+        SizeConstraints {
+            max: Size::new(width, height),
+            ..Self::DEFAULT
+        }
+    }
+}
+
+/// The space around and inside of a UI node
+///
+/// ## Margin
+///
+/// A margin is used to create space around UI elements, outside of any defined borders.
+///
+/// ```
+/// # use bevy_ui::{UiRect, Val, Spacing};
+/// #
+/// let margin = Spacing::margin(UiRect {
+///     left: Val::Px(30.0),
+///     right: Val::Px(25.0),
+///     top: Val::Px(10.0),
+///     bottom: Val::Px(20.0),
+/// });
+/// ```
+///
+/// ## Padding
+///
+/// A padding is used to create space around UI elements, inside of any defined borders.
+///
+/// ```
+/// # use bevy_ui::{UiRect, Val, Spacing};
+/// #
+/// let padding = Spacing::padding(UiRect {
+///     left: Val::Px(10.0),
+///     right: Val::Px(20.0),
+///     top: Val::Px(30.0),
+///     bottom: Val::Px(40.0),
+/// });
+/// ```
+///
+/// ## Borders
+///
+/// A border is used to define the width of the border of a UI element.
+///
+/// ```
+/// # use bevy_ui::{UiRect, Val, Spacing};
+/// #
+/// let border = Spacing::border(UiRect {
+///     left: Val::Px(10.0),
+///     right: Val::Px(20.0),
+///     top: Val::Px(30.0),
+///     bottom: Val::Px(40.0),
+/// });
+/// ```
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Spacing {
+    /// The space around the outside of the UI element
+    pub margin: UiRect,
+    /// The space around the inside of the UI element
+    pub padding: UiRect,
+    /// The space around the outside of the UI element that can be colored to create a visible border
+    pub border: UiRect,
```

Comment:
Should border be on its own `Border` component? I imagine that it eventually will need: `size`, `radius` and `color` (with independent values for each side/corner).

Reply:
Eventually yes. Details and implementation to be figured out later in a separate PR.

Reply:
Adding support for borders in Bevy would first require exposing borders upstream from Taffy, see https://github.com/DioxusLabs/taffy/issues/223

Reply:
Fixed borders could be implemented without that (simply read the border style to determine how thick the border is). Percentage borders would require taffy support though.